standalone/test/include/psen_scan_v2_standalone/util/integrationtest_helper.h

```diff
@@ -92,31 +101,54 @@ createValidMonitoringFrameMsg(const uint32_t scan_counter = 42,
 static std::vector<data_conversion_layer::monitoring_frame::Message>
 createValidMonitoringFrameMsgs(const uint32_t scan_counter, const std::size_t num_elements)
 {
-  std::vector<data_conversion_layer::monitoring_frame::Message> msgs(num_elements);
-  std::generate(msgs.begin(), msgs.end(), [scan_counter]() { return createValidMonitoringFrameMsg(scan_counter); });
+  std::vector<data_conversion_layer::monitoring_frame::Message> msgs;
+  std::generate_n(
+      std::back_inserter(msgs), num_elements, [scan_counter]() { return createValidMonitoringFrameMsg(scan_counter); });
   return msgs;
 }
 
 static std::vector<data_conversion_layer::monitoring_frame::Message>
 createMonitoringFrameMsgsForScanRound(const uint32_t scan_counter, const std::size_t num_elements)
 {
-  std::vector<data_conversion_layer::monitoring_frame::Message> msgs(num_elements);
-  for (std::size_t i = 0; i < msgs.size(); ++i)
+  std::vector<data_conversion_layer::monitoring_frame::Message> msgs;
+  for (std::size_t i = 0; i < num_elements; ++i)
   {
     const util::TenthOfDegree start_angle =
         (DEFAULT_SCAN_RANGE.getEnd() / static_cast<int>(num_elements)) * static_cast<int>(i);
     const util::TenthOfDegree end_angle =
         (DEFAULT_SCAN_RANGE.getEnd() / static_cast<int>(num_elements)) * static_cast<int>(i + 1);
-    msgs[i] = createValidMonitoringFrameMsg(scan_counter, start_angle, end_angle);
+    msgs.push_back(createValidMonitoringFrameMsg(scan_counter, start_angle, end_angle));
   }
   return msgs;
 }
 
+static std::vector<data_conversion_layer::monitoring_frame::MessageStamped>
+stampMonitoringFrameMsgs(const std::vector<data_conversion_layer::monitoring_frame::Message>& msgs, int64_t stamp)
+{
+  std::vector<data_conversion_layer::monitoring_frame::MessageStamped> stamped_msgs;
+  std::for_each(msgs.begin(), msgs.end(), [&stamped_msgs, stamp](const auto& msg) {
+    stamped_msgs.push_back(data_conversion_layer::monitoring_frame::MessageStamped(msg, stamp));
+  });
+  return stamped_msgs;
+}
+
 ACTION_P(OpenBarrier, barrier)
 {
   barrier->release();
 }
 
+MATCHER_P(ScanDataEqual, scan, "")
+{
+  return arg == scan;  // ToDo: misleading?!
```

Comment:
@rfeistenauer What do you think? Should the timestamp also be checked by the equality operator?

Reply:
Hmm jea the equality operator does only check measurement related things. 
We should probably test for the rest as well.

Is it intended, that the operator ignores Intensities, timestamp and  scan counter?

Reply:
I would move this topic to a separate p.r.